pithos/plugins/journald_logging.py

```diff
@@ -0,0 +1,144 @@
+        self._logger.addHandler(self._journal)
+        self._logging_changed_handler = self.preferences_dialog.connect('logging-changed', self._on_logging_changed)
+
+    def _on_logging_changed(self, prefs_dialog, level):
+        self.settings['data'] = level
+        self._journal.setLevel(LOG_LEVELS[level])
+        logging.info('setting journald logging level to: {}'.format(level))
+
+    def on_disable(self):
+        if self._logging_changed_handler:
+            self.preferences_dialog.disconnect(self._logging_changed_handler)
+        self._logger.removeHandler(self._journal)
+
+
+class LoggingPluginPrefsDialog(Gtk.Dialog):
+    __gtype_name__ = 'LoggingPluginPrefsDialog'
+    __gsignals__ = {
+        'logging-changed': (GObject.SignalFlags.RUN_FIRST, None, (GObject.TYPE_STRING,)),
+    }
+
+    def __init__(self, parent, settings, *args, **kwargs):
+        super().__init__(
+            _('Logging Level'),
+            parent,
+            0,
+            ('_Cancel', Gtk.ResponseType.CANCEL, '_Apply', Gtk.ResponseType.APPLY),
+            *args,
+            use_header_bar=1,
+            **kwargs,
+        )
+
+        self.settings = settings
+        self.set_resizable(False)
+        self.connect('response', self._on_response)
+
+        sub_title = Gtk.Label.new(_('Set the journald logging level for Pithos'))
+        sub_title.set_halign(Gtk.Align.CENTER)
+        self.log_level_combo = Gtk.ComboBoxText.new()
+
+        logging_levels = [
+            ('debug', 'High - debug'),
+            ('verbose', 'Default - verbose'),
+            ('warning', 'Low - warning'),
+        ]
+
+        for level in logging_levels:
+            self.log_level_combo.append(level[0], level[1])
+
+        self._reset_combo()
+        content_area = self.get_content_area()
+
+        content_area.add(sub_title)
+        content_area.add(self.log_level_combo)
+        content_area.show_all()
+
+    def _reset_combo(self):
+        self.log_level_combo.set_active_id(self.settings['data'] or 'verbose')
+
+    def _on_response(self, dialog, response):
+        if response != Gtk.ResponseType.APPLY:
+            self.hide()
+            self._reset_combo()
+            return
+
+        setting = self.settings['data']
+        active_id = self.log_level_combo.get_active_id()
+
+        if setting == active_id:
+            self.hide()
+            return
+
+        if active_id != 'debug':
+            self.hide()
+            self.emit('logging-changed', active_id)
+            return
+
+        def on_dialog_response(dialog, response):
+            if response == Gtk.ResponseType.YES:
+                self.hide()
+                self.emit('logging-changed', active_id)
+            dialog.destroy()
+
+        message = (_('The debug logging level is not '
+                   'recommended unless you are actually debugging an issue, '
+                   'as it generates very large logs.\n\nAre you sure you want to set logging to debug?'))
```

Comment:
__[flake8]__
* [E128] continuation line under-indented for visual indent

<!-- Comment by [SideCI](https://sideci.com) -->